File: src/language_binding.jl
#=
language_binding:
- Julia version: 
- Author: thinkAmi
- Date: 2019-09-17
=#

using HDF5

h5open("output/h5py/binding.h5", "r") do f
    dataset = read(f, "hello")
    print(dataset[1])  # Juliaのindexは1始まり
    # => ワールド
end

File: src/language_binding.py
# PyCon JP 2019 サンプル

import pathlib

import h5py

BASE_DIR = pathlib.Path(__file__).resolve().parents[0]

output = BASE_DIR.joinpath('output', 'h5py', 'binding.h5')
with h5py.File(output, mode='w') as file:
    dataset = file.create_dataset(
        name='hello', shape=(1,), dtype=h5py.string_dtype())
    dataset[0] = 'ワールド'
